### DISCUSSION TITLE
Opening Camera to send pictures in ios webview.

### DISCUSSION THREAD

**Discussion by farhazulMullick** [KevinnZou/compose-webview-multiplatform#115] [Q&A]
![File (1)](https://github.com/KevinnZou/compose-webview-multiplatform/assets/72504956/dc046ce3-b712-4d1b-ac9c-fc49aed1f224)

On clicking Take Photo or Video Xcode is throwing this error. 

`[CAMCaptureEngine] Received a session runtime error notification : Error Domain=AVFoundationErrorDomain Code=-11800 "The operation could not be completed" UserInfo={NSLocalizedFailureReason=An unknown error occurred (-12782), NSLocalizedDescription=The operation could not be completed, NSUnderlyingError=0x600000ded050 {Error Domain=NSOSStatusErrorDomain Code=-12782 "(null)"}}
[CAMCaptureEngine] Performing recovery from error: Error Domain=AVFoundationErrorDomain Code=-11800 "The operation could not be completed" UserInfo={NSLocalizedFailureReason=An unknown error occurred (-12782), NSLocalizedDescription=The operation could not be completed, NSUnderlyingError=0x600000ded050 {Error Domain=NSOSStatusErrorDomain Code=-12782 "(null)"}}
Snapshotting a view (0x103e64e20, CAMPreviewView) that has not been rendered at least once requires afterScreenUpdates:YES.
[CAMCaptureEngine] Attempting to recover from a session runtime error by restarting the AVCaptureSession...`

After that .

`03-07 13:42:02.963 💚 DEBUG Napier.d - WebView DisposableEffect ` is called.

Also granted camera and Microphone permissions.

**Comment by farhazulMullick** [comment]
@KevinnZou Any help.

**Reply by KevinnZou** [reply]
@farhazulMullick Thanks for your feedback! Does that work for Android? Could you provide a test URL?

**Reply by farhazulMullick** [reply]
@KevinnZou Don't have testing web url. Was testing this on confidential url. We can load a sample html content which accepts file.
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>File Input Example</title>
</head>
<body>

    <form action="/upload" method="post" enctype="multipart/form-data">
        <label for="fileInput">Choose a file:</label>
        <input type="file" id="fileInput" name="fileInput" accept=".jpg, .jpeg, .png" multiple>
        <br>
        <input type="submit" value="Upload">
    </form>

</body>
</html>
```

**Reply by farhazulMullick** [reply]
@KevinnZou This issue is already reported over [compose-multiplatform](https://github.com/JetBrains/compose-multiplatform/issues/3779). 
May be you can cross verify.

**Reply by KevinnZou** [reply]
@farhazulMullick Thanks for your information! I tested it with [file.io](www.file.io) on my device. It successfully launches the camera and allows me to choose a photo. However, when I confirm the photo and return to the web page, it refreshes and loses its state. This indicates that there is still a problem with the image picker.

After conducting research, I discovered a similar issue under the compose-multiplatform repository, which can be found at [this link](https://github.com/JetBrains/compose-multiplatform/issues/3698). This issue appears to be related to the dispose strategy of the CMP and can only be resolved on a case-by-case basis. Therefore, I suggest that we raise this issue with compose multiplatform and await their proposed solutions.
